src/adagrams.js

```diff
@@ -1,15 +1,172 @@
+const LETTER_POOL = {
+    A: 9, 
+    B: 2, 
+    C: 2, 
+    D: 4, 
+    E: 12, 
+    F: 2, 
+    G: 3, 
+    H: 2, 
+    I: 9, 
+    J: 1, 
+    K: 1, 
+    L: 4, 
+    M: 2, 
+    N: 6, 
+    O: 8, 
+    P: 2, 
+    Q: 1, 
+    R: 6, 
+    S: 4, 
+    T: 6, 
+    U: 4, 
+    V: 2, 
+    W: 2, 
+    X: 1, 
+    Y: 2, 
+    Z: 1,
+}
+
+const LETTER_VALUE = {
+    A: 1, 
+    B: 3, 
+    C: 3, 
+    D: 2, 
+    E: 1, 
+    F: 4, 
+    G: 2, 
+    H: 4, 
+    I: 1, 
+    J: 8, 
+    K: 5, 
+    L: 1, 
+    M: 3, 
+    N: 1, 
+    O: 1, 
+    P: 3, 
+    Q: 10, 
+    R: 1, 
+    S: 1, 
+    T: 1, 
+    U: 1, 
+    V: 4, 
+    W: 4, 
+    X: 8, 
+    Y: 4, 
+    Z: 10,
+}
+
+const getRandomInt = (max) => {
+  return Math.floor(Math.random() * max);
+}
+
 export const drawLetters = () => {
-  // Implement this method for wave 1
+  const alphabet = Object.keys(LETTER_POOL);
+  const drawnLetters = [];
+
+  while (drawnLetters.length < 10) {
+    const randomLetter = alphabet[getRandomInt(26)];
+    const quantity = LETTER_POOL[randomLetter]; 
+    let counter = 0;
+
+    for (let i = 0; i < drawnLetters.length; i++) {
+      if (drawnLetters[i] === randomLetter) {
+        counter += 1;
+      } 
+    }
+    if (counter < quantity) {
+      drawnLetters.push(randomLetter);
+    }
```

Comment:
This doesn't quite mimic the action of pulling out a random letter form a bag of properly weighted letters (here, there's an equal chance of pulling out a `E` as a `Z` due to L28, despite `E` being much more likely to be drawn) but it's a good start!